### PR TITLE
Feature/112823 exception form

### PR DESCRIPTION
**Pull request by nealfennimore**
https://central.tri.be/issues/112823

Depends on #275 

- Adds recursive block nesting in https://github.com/moderntribe/events-gutenberg/commit/da0cf009353477cbd2d7d4552740ef491a64608a
- Adds removal and add flows for fields
- Sets up initial data for forms

### CODE REVIEW

**Comment by mitogh** [comment]
Some conflicts and tests outdated or failing it seems.

File: plugins/events-pro/src/modules/elements/row/element.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import './style.pcss';
+
+const Row = ( { children, className } ) => {
+	return (
```
Comment (mitogh) [inline plugins/events-pro/src/modules/elements/row/element.js]: Seems like this one can be removed.

**Comment by nealfennimore** [comment]
Good to :eye:

File: plugins/events-pro/src/modules/data/blocks/exception/reducer.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+export default function exception( state = [], action ) {
+	switch ( action.type ) {
+		case types.ADD_EXCEPTION_FIELD:
+			return [
+				...state,
+				action.payload,
+			];
+		case types.REMOVE_EXCEPTION_FIELD:
+			return state.filter( field => field.id === action.payload.id );
```
Comment (paulmskim) [inline plugins/events-pro/src/modules/data/blocks/exception/reducer.js]: Should this be `!==`?

File: plugins/events-pro/src/modules/data/blocks/recurring/reducer.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+export default function recurring( state = [], action ) {
+	switch ( action.type ) {
+		case types.ADD_RECURRING_FIELD:
+			return [
+				...state,
+				action.payload,
+			];
+		case types.REMOVE_RECURRING_FIELD:
+			return state.filter( field => field.id === action.payload.id );
```
Comment (paulmskim) [inline plugins/events-pro/src/modules/data/blocks/recurring/reducer.js]: Also `!==` here?

**Comment by nealfennimore** [comment]
:moar: :eyes: please

File: plugins/events-pro/src/modules/elements/recurring-field/singular.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+						value="single"
+					/>
+				</div>
+			</Row>
+			<Row>
+				<Label>
+					{ __( 'On', 'events-gutenberg' ) }
+				</Label>
+				<div>
+					<DatePicker datetime="1999-01-01 12:00:00" />
```
[the remark this answers is not in the document]
Comment (nealfennimore) [reply plugins/events-pro/src/modules/elements/recurring-field/singular.js]: Yep WIP

File: plugins/events-pro/src/modules/elements/recurring-field/singular.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	return (
+		<Fragment>
+			<Row>
+				<Label>
+					{ __( 'A', 'events-gutenberg' ) }
+				</Label>
+				<div>
+					<Select
+						options={ options.RECURRENCE_TYPE_RULES_OPTIONS }
+						value="single"
```
[the remark this answers is not in the document]
Comment (nealfennimore) [reply plugins/events-pro/src/modules/elements/recurring-field/singular.js]: this isn't done yet - data isn't hooked in. Just placeholders for now

File: plugins/events-pro/src/modules/elements/label/element.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+ * External Dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import './style.pcss';
+
+const Label = ( { children, className } ) => (
+	<div className={ classnames( 'tribe-events-pro__label', className ) }>
+		<span>{ children }</span>
```
[the remark this answers is not in the document]
Comment (nealfennimore) [reply plugins/events-pro/src/modules/elements/label/element.js]: This ends up being treated as a flex item due to the parent div, so it should be okay here

File: plugins/events/src/modules/blocks/event-datetime/hooks/blocks.js [hunk 7]
```diff
@@ -12,7 +12,10 @@ import { PluginBlockHooks } from '@moderntribe/common/components';
 
 const PLUGIN_TEMPLATES = {
 	[ constants.EVENTS_PRO_PLUGIN ]: [
-		[ 'tribe/event-pro-recurring', {}],
+		[ 'tribe/event-pro-recurring-entry', {}, [
```
[the remark this answers is not in the document]
Comment (nealfennimore) [reply plugins/events/src/modules/blocks/event-datetime/hooks/blocks.js]: Cool - I'll update

File: plugins/events-pro/src/modules/elements/recurring-field/singular.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (bordoni) [reply plugins/events-pro/src/modules/elements/recurring-field/singular.js]: 💟

File: plugins/events-pro/src/modules/elements/recurring-field/singular.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (bordoni) [reply plugins/events-pro/src/modules/elements/recurring-field/singular.js]: 💟

File: plugins/events-pro/src/modules/elements/label/element.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (bordoni) [reply plugins/events-pro/src/modules/elements/label/element.js]: 💟

**Comment by nealfennimore** [comment]
winner winner :chicken: :plate_with_cutlery: ? More :eyes: please

**Pull request merged**